transformer_engine/jax/cpp_extensions.py

```diff
@@ -1979,6 +1981,23 @@ def scaled_upper_triang_masked_softmax_bwd(grad_outputs: jnp.ndarray, softmax_ou
                                                           scale_factor=scale_factor)
 
 
+def _check_rng_state(rng_state):
+    # Jax can't bind None, create a dummy tensor for None
```

Comment:
Can you confirm how this work?
If end user don't specify an rng, but set a dropout rate, we will generate a static seed?
If the user end up creating many such layers, all the dropout seed will end up being the same.
Is that right?

I don't like that. What about having a global rng that we split everything we need one?
This way, it will still be deterministic, but each layers will have a different seed everytimes.

Reply:
The condition rng_state=None should only occur when the dropout rate is zero., I will add an assertion for this case. So, the dummy tensor will only be created when the dropout rate is zero. This creation is necessary for JAX custom call binding, as JAX cannot bind a None tensor. This dummy tensor will not be used in the underlying kernels since the dropout rate is zero.

Reply:
Enhance the rng_state checking in https://github.com/NVIDIA/TransformerEngine/commit/c8615cf3866fa0266d20739ef221b1ccdb5e45ea. I add the check that the `rng_state` must not be `None` if dropout is enabled